src/main/java/com/minecolonies/rankup/util/AccountingUtils.java

```diff
@@ -0,0 +1,568 @@
+package com.minecolonies.rankup.util;
+
+import com.minecolonies.rankup.Rankup;
+import com.minecolonies.rankup.modules.core.config.AccountConfigData;
+import com.minecolonies.rankup.modules.databases.DatabaseModule;
+import org.spongepowered.api.Sponge;
+import org.spongepowered.api.entity.living.player.User;
+import org.spongepowered.api.service.sql.SqlService;
+import org.spongepowered.api.service.user.UserStorageService;
+import uk.co.drnaylor.quickstart.exceptions.NoModuleException;
+
+import javax.sql.DataSource;
+import java.sql.*;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.UUID;
+
+import static com.minecolonies.rankup.util.Constants.SQL.*;
+
+/**
+ * Lots of accounts based utils (Used for database / file config adaption)
+ */
+public class AccountingUtils extends ConfigUtils
+{
+
+    private SqlService sql;
+    private Connection conn;
+
+    private              String tableId            = TABLE_SUFFIX;
+    private static final String TABLE_SUFFIX       = "player_stats";
+    private static final String UUID_COLUMN        = "UUID";
+    private static final String PLAYER_NAME_COLUMN = "PLAYER_NAME";
+    private static final String JOIN_DATE_COLUMN   = "JOIN_DATE";
+    private static final String LAST_JOIN_COLUMN   = "LAST_JOIN";
+    private static final String TIME_PLAYED_COLUMN = "TIME_PLAYED";
+
+    public AccountingUtils(final Rankup pl)
+    {
+        super(pl);
+    }
+
+    @SuppressWarnings("squid:S2259")
+    private DataSource getDataSource(String jdbcUrl) throws SQLException
+    {
+        tableId = getDatabasesConfig().sqlTablePrefix + TABLE_SUFFIX;
+        if (sql == null)
+        {
+            sql = Sponge.getServiceManager().provide(SqlService.class).orElse(null);
+        }
+        return sql.getDataSource(jdbcUrl);
+    }
+
+    private String getColumn(final String column)
+    {
+        switch (column.toUpperCase())
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Define the locale to be used in this String operation. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orionminecraft.com/coding_rules#rule_key=squid%3AS1449)